test/sink.test.js

```diff
@@ -85,6 +91,30 @@ test('.writer() - no value for "type" argument - should throw', () => {
     }).toThrowError('"type" is missing');
 });
 
+test('directory() - should fetch file in root directory', async () => {
+    const sink = new Sink({ path: './test/mock' });
+    const directoryContent = await sink.directory('/');
+    expect(directoryContent.length).toBe(1);
+    expect(directoryContent[0].fileName).toMatchSnapshot();
+    expect(directoryContent[0].content.length).toMatchSnapshot();
+});
+
+test('directory() - should fetch all files in sub directory', async () => {
+    const sink = new Sink({ path: './test/mock/' });
+    const directoryContent = await sink.directory('/sub');
+    expect(directoryContent).toMatchSnapshot();
+});
+
+test('directory() - should error when directory is missing', async () => {
```

Comment:
Should it error?

Reply:
I wondered that too. I think as long as we are consistent across syncs and document the behaviour we should be good.

Reply:
perhaps lean towards erroring with clear usage fail message. Thats probably clearer then wondering what it does if you don't give it a path (as a consumer)